packages/core/database/src/metadata/__tests__/resources/models.ts

```diff
@@ -4352,128 +181,45 @@ export const models = [
-      },
-      name: {
-        type: 'string',
-        required: true,
-      },
-    },
-  },
-  {
-    uid: 'blog.test-como',
-    singularName: 'test-como',
-    tableName: 'components_blog_test_comos',
-    attributes: {
-      id: {
-        type: 'increments',
-      },
-      name: {
-        type: 'string',
-        default: 'toto',
-      },
-    },
-  },
-  {
-    uid: 'basic.simple',
-    singularName: 'simple',
-    tableName: 'components_basic_simples',
-    attributes: {
-      id: {
-        type: 'increments',
-      },
-      name: {
-        type: 'string',
-        required: true,
-      },
-      test: {
-        type: 'string',
-      },
-    },
-  },
-  {
-    uid: 'basic.relation',
-    singularName: 'relation',
-    tableName: 'components_basic_relations',
-    attributes: {
-      id: {
-        type: 'increments',
-      },
-      categories: {
+    dynamicZone: {
+      dz: {
         type: 'relation',
-        relation: 'oneToMany',
-        target: 'api::category.category',
-      },
-    },
-  },
-  {
-    uid: 'strapi::core-store',
-    singularName: 'strapi_core_store_settings',
-    tableName: 'strapi_core_store_settings',
-    attributes: {
-      id: {
-        type: 'increments',
-      },
-      key: {
-        type: 'string',
-      },
-      value: {
-        type: 'text',
-      },
-      type: {
-        type: 'string',
-      },
-      environment: {
-        type: 'string',
-      },
-      tag: {
-        type: 'string',
-      },
-    },
-  },
-  {
-    uid: 'strapi::webhook',
-    singularName: 'strapi_webhooks',
-    tableName: 'strapi_webhooks',
-    attributes: {
-      id: {
-        type: 'increments',
-      },
-      name: {
-        type: 'string',
-      },
-      url: {
-        type: 'text',
-      },
-      headers: {
-        type: 'json',
-      },
-      events: {
-        type: 'json',
-      },
-      enabled: {
-        type: 'boolean',
+        relation: 'morphToMany',
+        joinTable: {
+          name: 'complexes_components',
+          joinColumn: {
+            name: 'entity_id',
+            referencedColumn: 'id',
+          },
+          morphColumn: {
+            idColumn: {
+              name: 'component_id',
+              referencedColumn: 'id',
+            },
+            typeColumn: {
+              name: 'component_type',
+            },
+            typeField: '__component',
+          },
+          on: {
+            field: 'dz',
+          },
+          orderBy: {
+            order: 'asc',
+          },
+          pivotColumns: ['entity_id', 'component_id', 'field', 'component_type'],
+        },
       },
     },
   },
-];
+};
+
+type ModelAttributes<T = typeof attributes> = {
+  [P in keyof T]: T[P] extends { [key: string]: infer U } ? U : never;
+}[keyof T];
+
+export const buildModelWith = (attributes: ModelAttributes) => {
+  const model = { ...baseModel };
+  model.attributes = { ...model.attributes, ...attributes };
+  return model;
+};
```

Comment:
What are we doing here? Creating a model with additional attributes?

Reply:
@christiancp100 added this so maybe he can clarify, but yeah, this is basically a test utility to simplify building models to be tested